File: subprojects/language-web/src/main/java/tools/refinery/language/web/api/dto/OutputFormats.java
/*
 * SPDX-FileCopyrightText: 2025 The Refinery Authors <https://refinery.tools/>
 *
 * SPDX-License-Identifier: EPL-2.0
 */
package tools.refinery.language.web.api.dto;

import jakarta.validation.Valid;
import jakarta.validation.constraints.NotNull;

public final class OutputFormats {
	@NotNull
	@Valid
	private JsonOutputFormat json = new JsonOutputFormat();

	@NotNull
	@Valid
	private SourceOutputFormat source = new SourceOutputFormat();

	public JsonOutputFormat getJson() {
		return json;
	}

	public void setJson(JsonOutputFormat json) {
		this.json = json;
	}

	public SourceOutputFormat getSource() {
		return source;
	}

	public void setSource(SourceOutputFormat source) {
		this.source = source;
	}
}

File: subprojects/client-js/src/dto/OutputFormats.ts
/*
 * SPDX-FileCopyrightText: 2025 The Refinery Authors <https://refinery.tools/>
 *
 * SPDX-License-Identifier: EPL-2.0
 */

import z from 'zod/v4';

const OutputFormat = z.object({
  enabled: z.boolean().default(true),
});

export const PartialInterpretationPreservation = z.enum(['keep', 'discard']);

export type PartialInterpretationPreservation = z.infer<
  typeof PartialInterpretationPreservation
>;

export const JsonOutputFormat = OutputFormat.extend({
  nonExistingObjects: PartialInterpretationPreservation.default('discard'),
  shadowPredicates: PartialInterpretationPreservation.default('discard'),
});

export type JsonOutputFormat = z.infer<typeof JsonOutputFormat>;

export const SourceOutputFormat = OutputFormat;

export type SourceOutputFormat = z.infer<typeof SourceOutputFormat>;

export const OutputFormats = z
  .object({
    json: JsonOutputFormat.prefault({}),
    source: SourceOutputFormat.prefault({}),
  })
  .prefault({});

export type OutputFormats = z.infer<typeof OutputFormats>;
